pkgs/applications/misc/navit/default.nix

```diff
@@ -21,34 +31,54 @@ stdenv.mkDerivation rec {
     sha256 = "0vg6b6rhsa2cxqj4rbhfhhfss71syhnfa6f1jg2i2d7l88dm5x7d";
   };
 
-  #hardeningDisable = [ "format" ];
-  NIX_CFLAGS_COMPILE = [ "-I${SDL.dev}/include/SDL" ];
+  patches = [ ./CMakeLists.txt.patch ];
+  hardeningDisable = [ "format" ];
+
+  NIX_CFLAGS_COMPILE = [ ]
+    ++ optional sdlSupport "-I${SDL.dev}/include/SDL"
+    ++ optional speechdSupport "-I${speechd}/include/speech-dispatcher";
 
-  # TODO: fix speech options.
-  cmakeFlags = [ "-DSAMPLE_MAP=n " "-DCMAKE_BUILD_TYPE=RelWithDebInfo" "-Dsupport/espeak=FALSE" "-Dspeech/qt5_espeak=FALSE" ];
+  # we choose only cmdline and speech-dispatcher speech options.
+  # espeak builtins is made for non-cmdline OS as winCE
+  cmakeFlags = [
+    "-DSAMPLE_MAP=n " "-DCMAKE_BUILD_TYPE=RelWithDebInfo"
+    "-Dspeech/qt5_espeak=FALSE" "-Dsupport/espeak=FALSE"
+  ];
 
-  buildInputs = [ gtk2 SDL fontconfig freetype imlib2 SDL_image libGLU_combined freeimage libxslt
-    libXmu freeglut python gettext quesoglc gd postgresql qtbase SDL_ttf fribidi pcre qtquickcontrols
-    espeak qtmultimedia qtspeech qtsensors qtlocation qtdeclarative qtsvg dbus-glib librsvg shapelib glib 
-    cairo gdk_pixbuf pango atk ];
+  buildInputs = [
+    gtk2 fontconfig freetype imlib2 libGLU_combined freeimage
+    libxslt libXmu freeglut python gettext quesoglc gd
+    fribidi pcre  dbus dbus-glib librsvg shapelib glib
+    cairo gdk_pixbuf pango atk
+  ] ++ optionals sdlSupport [ SDL SDL_ttf SDL_image ]
+    ++ optional postgresqlSupport postgresql
+    ++ optional speechdSupport speechd
+    ++ optionals qtSupport [
+      qtquickcontrols qtmultimedia qtspeech qtsensors
+      qtbase qtlocation qtdeclarative qtsvg
+  ];
 
-  nativeBuildInputs = [ pkgconfig cmake patchelf bzip2 ];
+  nativeBuildInputs = [ makeWrapper pkgconfig cmake patchelf bzip2 ];
 
   # we dont want blank screen by defaut
   postInstall = ''
+    # copy a better configuration for nixos desktop
+    cp "${./navit.xml}" $out/share/navit/
     # emulate DSAMPLE_MAP
-    mkdir -p $out/share/navit/maps/maps
+    mkdir -p $out/share/navit/maps/
```

Comment:
Please quote `$out`

Reply:
Quoting is not necessary. We do not support spaces in nix store path. Even if would fix all our code we cannot fix all upstream projects. Therefore we keep code simple and do not quote at all.

Reply:
I didn't know that we do not support that, do you have a source on that? I agree that it would probably lead to a multitude of problems and be impractical at the moment, but that doesn't mean that we shouldn't try to do the "right" thing as much as possible.

I also disagree that leaving out the quotes makes the code simpler. It makes it necessary to reason about weather or not this variable may have spaces. It is far simpler in my opinion to always quote variables except in the cases where splitting on whitespace is explicitly wanted. Otherwise it always comes back to bite you in the future. Also quoting unconditionally would allow us to run shellcheck on scripts, as was proposed somewhere before.

So I don't see any disadvantage to quoting but quite some advantages.

Reply:
Okay, good to know. Thanks. My other points are still valid though.

Reply:
At the moment omitting quotation is common sense:

```console
$ ag 'wrapProgram $out' | wc -l
596
$ ag 'wrapProgram "$out' | wc -l
179
```

There is also sometimes pitfalls of quoting twice.

If you want to change direction of how we handle quoting in builder scripts it is better to open a new issue. Because otherwise our reviews will become inconsistent which can be annoying for contributors.

Reply:
I don't think "the way it is commonly done" and "common sense" are necessarily the same thing.

Do do that consistently we would need a style guide and automated tooling. Thats a way bigger problem to tackle.